mobile/lib/ui/viewer/gallery/empty_album_state_new.dart

```diff
@@ -0,0 +1,134 @@
+import "package:fast_base58/fast_base58.dart";
+import "package:flutter/material.dart";
+import "package:photos/generated/l10n.dart";
+import "package:photos/models/api/collection/public_url.dart";
+import "package:photos/models/collection/collection.dart";
+import "package:photos/models/selected_files.dart";
+import "package:photos/services/collections_service.dart";
+import "package:photos/theme/ente_theme.dart";
+import "package:photos/ui/components/buttons/button_widget.dart";
+import "package:photos/ui/components/models/button_type.dart";
+import "package:photos/ui/viewer/gallery/hooks/add_photos_sheet.dart";
+import "package:photos/utils/dialog_util.dart";
+import "package:photos/utils/share_util.dart";
+
+class EmptyAlbumStateNew extends StatefulWidget {
+  final Collection c;
+  final SelectedFiles? selectedFiles;
+  const EmptyAlbumStateNew(
+    this.c, {
+    super.key,
+    this.selectedFiles,
+  });
+
+  @override
+  State<EmptyAlbumStateNew> createState() => _EmptyAlbumStateNewState();
+}
+
+class _EmptyAlbumStateNewState extends State<EmptyAlbumStateNew> {
+  final ValueNotifier<bool> _hasSelectedFilesNotifier = ValueNotifier(false);
+
+  @override
+  void initState() {
+    super.initState();
+    widget.selectedFiles?.addListener(_selectedFilesListener);
+  }
+
+  @override
+  void dispose() {
+    _hasSelectedFilesNotifier.dispose();
+    widget.selectedFiles!.removeListener(_selectedFilesListener);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final GlobalKey shareLinkAlbumButtonKey = GlobalKey();
```

Comment:
[GlobalKeys should not be re-created on every build. They should usually be long-lived objects owned by a State object](https://api.flutter.dev/flutter/widgets/GlobalKey-class.html)